x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.test.ts

```diff
@@ -179,7 +180,106 @@ describe.each([
       });
     });
 
-    it('returns partial failure error if couple of rule validations fail and the rest are successfull', async () => {
+    it('returns error if index patterns action is applied to machine learning rule', async () => {
+      readRulesMock.mockImplementationOnce(() =>
+        Promise.resolve({ ...mockRule, params: { ...mockRule.params, type: 'machine_learning' } })
+      );
+
+      const request = requestMock.create({
+        method: 'patch',
+        path: DETECTION_ENGINE_RULES_BULK_ACTION,
+        body: {
+          ...getPerformBulkActionEditSchemaMock(),
+          ids: ['failed-mock-id'],
+          query: undefined,
+          edit: [
+            {
+              type: 'add_index_patterns',
```

Comment:
Should we also test `delete_index_patterns` and `set_index_patterns`?

Reply:
I think it's redundant, these 2 were tested in unit tests within `x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_edit.test.ts`